contracts/protocol/libs/FundsLib.sol

```diff
@@ -276,6 +425,14 @@ library FundsLib {
         // first decrease the amount to prevent the reentrancy attack
         decreaseAvailableFunds(_entityId, _tokenAddress, _amount);
 
+        // try to transfer the funds
+        transferFundsFromProtocol(_tokenAddress, _to, _amount);
+
+        // notify the external observers
+        emit FundsWithdrawn(_entityId, _to, _tokenAddress, _amount, EIP712Lib.msgSender());
```

Comment:
Yes, that was intended, since not all transfers correspond to the withdrawals (for example returning the surplus).

But maybe you are right, currently just looking at events is not enough to have the full picture of the funds:
- if the buyer escalates the dispute, they need to pay the `buyerEscalationDeposit`. And we do not emit any `FundsDeposited` or `FundsEncumbered`.
- In `sequentialCommitToOffer` we should also probably emit `FundsEncumbered`, at least for the part that is added to the escrow.

I'll need to investigate this a bit more before making any changes, but most likely I'll add the relevant events in few places.

Reply:
I added the relevant events now. With the combination of `FundsDeposited` (sellers only), `FundsEncumbered`, `FundsReleased` and `FundsWithdrawn` I believe it's possible to get the full picture in the escrow.

The only bit of history that is missing is if someone already escalated the dispute. But, given that currently the `buyerEscalationDeposit` is 0, it is not problematic.